applications/Anchor.md

```diff
@@ -0,0 +1,286 @@
+# Anchor, On-chain Linked List pallet and Name Service
+
+- **Team Name:** Fuu
+- **Payment Address:** 0xF8105ea1cC3130501bad5De60Bd3D794a9115dE2 (USDT)
+- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 1
+
+
+## Project Overview :page_facing_up:
+
+### Overview
+
+- Anchor is a on-chain linked list system base on substrate 3.0.0. It is used to support cApp ( Chain Application ) development. On another hand, Anchor can alse be treated as Name Service or on-chain key-value storage system.
+
+- Anchor is a isolated substrate pallet. It is currently used in the application of freeSaying. It can provide flexible data structure on the chain and handle complex logic without upgrading the substrate node itself.
+You can access the [freeSaying ( Only applicable to mobile )](https://freesaying.net) demo to know well. Anchor network is the very important basic storage system. Anchor.js is a isolate JS library to access Anchor network, can read and write data easily.
+Anchor pallet is a part of EasyPolka, the relationship as follow :
+
+![easypolka.png](http://android.im/anchor/easypolka.png)
```

Comment:
What are `cSaying` and `cHome` here?

Reply:
> What are `cSaying` and `cHome` here?

Hello @semuelle , sorry for the slow response. "cSaying" and "cHome" is cApp ( Chain Application ) which are the pure JS code stored on-chain by Anchor.
For example, such JS code can be launched by "saying" that the plinth for cApp.

```JSON
/* cApp basic structure.
*   @param RPC          object      //RPC object from plinth
*   @param container    string      //html DOM id for the cApp as container
*   @param error        object      //Error object from plinth
*/
;(function(RPC,container,error){
    // normal console log.
    console.log("Hello world, this is from cApp.");
    console.log(`Dom container id : ${container}`);
    console.log(`Error from pedestal : ${JSON.stringify(error)}`);

   // call RPC from plinth to access Anchor network
    var ss58="5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY";
    RPC.common.balance(ss58,function(res){
        console.log(res);
    });

})(RPC,domID,error);
```

So the data sent to setAnchor will be like this :

```JSON
{
  "key":"any_name",
  "raw":"JS string above",
  "protocol": JSON.stringify({"type":"app","format":"JS","ver":"0.0.1"}),
  "pre":0
}
```

Reply:
> What are `cSaying` and `cHome` here?

Hello @semuelle , sorry for the slow response. "cSaying" and "cHome" is cApp ( Chain Application ) which are the pure JS code stored on-chain by Anchor.
For example, such JS code can be launched by "saying" that the plinth for cApp.

```Javascript
/* cApp basic structure.
*   @param RPC          object      //RPC object from plinth
*   @param container    string      //html DOM id for the cApp as container
*   @param error        object      //Error object from plinth
*/
;(function(RPC,container,error){
    // normal console log.
    console.log("Hello world, this is from cApp.");
    console.log(`Dom container id : ${container}`);
    console.log(`Error from pedestal : ${JSON.stringify(error)}`);

   // call RPC from plinth to access Anchor network
    var ss58="5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY";
    RPC.common.balance(ss58,function(res){
        console.log(res);
    });

})(RPC,domID,error);
```

So the data sent to setAnchor will be like this :

```Javascript
{
  "key":"any_name",
  "raw":"JS string above",
  "protocol": JSON.stringify({"type":"app","format":"JS","ver":"0.0.1"}),
  "pre":0
}
```